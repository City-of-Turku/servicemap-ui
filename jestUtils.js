import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });