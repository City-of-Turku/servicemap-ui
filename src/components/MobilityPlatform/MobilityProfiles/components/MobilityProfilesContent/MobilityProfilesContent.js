import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@mui/material';
import { useIntl } from 'react-intl';
import { ReactSVG } from 'react-svg';
import { css } from '@emotion/css';
import moose from 'servicemap-ui-turku/assets/icons/contrast/icons-icon_moose-bw-v2.svg';
import fox from 'servicemap-ui-turku/assets/icons/contrast/icons-icon_fox-bw-v2.svg';
import deer from 'servicemap-ui-turku/assets/icons/contrast/icons-icon_deer-bw-v2.svg';
import rabbit from 'servicemap-ui-turku/assets/icons/contrast/icons-icon_rabbit-bw-v2.svg';
import marten from 'servicemap-ui-turku/assets/icons/contrast/icons-icon_marten-bw-v2.svg';
import capercaillie from 'servicemap-ui-turku/assets/icons/contrast/icons-icon_capercaillie-bw-v2.svg';
import useLocaleText from '../../../../../utils/useLocaleText';
import {
  StyledContainer,
  StyledFlexContainer,
  StyledHeaderContainer,
  StyledTextContainer,
} from '../../../styled/styled';

const MobilityProfilesContent = ({ postcodeArea, mobilityProfiles }) => {
  const intl = useIntl();
  const getLocaleText = useLocaleText();

  const filteredMobilityProfiles = mobilityProfiles.filter(
    item => item.postal_code_string === postcodeArea.name.fi && item.postal_code_type_string === 'Home' && item.count >= 1,
  );

  const iconClass = css({
    width: '30px',
    height: '30px',
  });

  const getIconByTopic = resultNum => {
    switch (resultNum) {
      case 1:
        return moose;
      case 2:
        return fox;
      case 3:
        return rabbit;
      case 4:
        return marten;
      case 5:
        return deer;
      case 6:
        return capercaillie;
      default:
        return moose;
    }
  };

  const noResultsText = () => (
    <StyledTextContainer>
      <Typography variant="body2" component="p">
        {intl.formatMessage({ id: 'area.mobilityResults.empty' })}
      </Typography>
    </StyledTextContainer>
  );

  return (
    <StyledContainer>
      <StyledHeaderContainer>
        <Typography variant="subtitle1" component="p">
          {intl.formatMessage({ id: 'area.mobilityResults.postCodeArea' }, { value: postcodeArea.name.fi })}
        </Typography>
      </StyledHeaderContainer>
      {filteredMobilityProfiles?.length
        ? filteredMobilityProfiles.map(item => (
          <StyledFlexContainer key={item.id}>
            <StyledTextContainer>
              <ReactSVG src={getIconByTopic(item.result)} className={iconClass} />
            </StyledTextContainer>
            <StyledTextContainer>
              <Typography variant="body2" component="p">
                {getLocaleText(item.result_topics)}
              </Typography>
            </StyledTextContainer>
            <StyledTextContainer>
              <Typography variant="body2" component="p">
                {item.count}
              </Typography>
            </StyledTextContainer>
          </StyledFlexContainer>
        ))
        : noResultsText()}
    </StyledContainer>
  );
};

MobilityProfilesContent.propTypes = {
  postcodeArea: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.shape({
      fi: PropTypes.string,
    }),
  }),
  mobilityProfiles: PropTypes.arrayOf(
    PropTypes.shape({
      postal_code_string: PropTypes.string,
      postal_code_type_string: PropTypes.string,
      result: PropTypes.number,
    }),
  ),
};

MobilityProfilesContent.defaultProps = {
  postcodeArea: {},
  mobilityProfiles: [],
};

export default MobilityProfilesContent;
