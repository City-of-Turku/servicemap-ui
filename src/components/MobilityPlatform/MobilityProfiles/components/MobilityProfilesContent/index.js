import MobilityProfilesContent from './MobilityProfilesContent';

export default MobilityProfilesContent;
