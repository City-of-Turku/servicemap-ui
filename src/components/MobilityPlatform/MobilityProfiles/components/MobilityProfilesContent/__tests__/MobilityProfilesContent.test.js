// Link.react.test.js
import React from 'react';
import MobilityProfilesContent from '../index';
import { initialState } from '../../../../../../redux/reducers/user';
import finnishTranslations from '../../../../../../i18n/fi';
import { getRenderWithProviders } from '../../../../../../../jestUtils';

const mockProps = {
  postcodeArea: {
    name: {
      en: '20210',
      fi: '20210',
      sv: '20210',
    },
  },
  mobilityProfiles: [
    {
      postal_code_string: '20210',
      postal_code_type_string: 'Home',
      result: 3,
      result_topics: {
        fi: 'Joustava JÃ¤nis',
        sv: 'Hinderfri Hare',
        en: 'Hassle-free Hare',
      },
      count: 10,
    },
  ],
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
});

describe('<MobilityProfilesContent />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<MobilityProfilesContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<MobilityProfilesContent {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toContain(`${finnishTranslations['area.mobilityResults.postCodeArea'].replace(
      '{value}',
      mockProps.postcodeArea.name.fi,
    )}`);
    expect(p[1].textContent).toContain(mockProps.mobilityProfiles[0].result_topics.fi);
    expect(p[2].textContent).toContain(mockProps.mobilityProfiles[0].count);
  });
});
