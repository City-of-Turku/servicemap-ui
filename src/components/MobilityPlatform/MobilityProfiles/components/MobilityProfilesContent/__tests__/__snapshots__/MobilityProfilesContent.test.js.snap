// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobilityProfilesContent /> should match snapshot 1`] = `
<div>
  <div
    class="css-1ynyhby"
  >
    <div
      class="css-1dznbe2"
    >
      <p
        class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
      >
        Postinumeroalue: 20210
      </p>
    </div>
    <div
      class="css-1do907m"
    >
      <div
        class="css-1fhgjcy"
      >
        <div
          class="css-c2uxn6"
        >
          <div>
            <div
              data-src="test-file-stub"
              src=""
            />
          </div>
        </div>
      </div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          Joustava JÃ¤nis
        </p>
      </div>
      <div
        class="css-1fhgjcy"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          10
        </p>
      </div>
    </div>
  </div>
</div>
`;
